lib/bitcoin/electrum_wallet.dart

```diff
@@ -174,28 +172,75 @@ abstract class ElectrumWalletBase extends WalletBase<ElectrumBalance,
       }
     }
 
-    if (inputs.isEmpty ||
-        (credentialsAmount > 0 && allInputsAmount < credentialsAmount)) {
+    if (inputs.isEmpty) {
       throw BitcoinTransactionNoInputsException();
     }
 
-    final allAmountFee =
-      feeAmountForPriority(transactionCredentials.priority, inputs.length, 1);
+    final allAmountFee = feeAmountForPriority(
+        transactionCredentials.priority, inputs.length, sendItemList.length);
     final allAmount = allInputsAmount - allAmountFee;
 
-    final amount = transactionCredentials.amount == null ||
-        allAmount - credentialsAmount < minAmount
-        ? allAmount
-        : credentialsAmount;
-    final fee = transactionCredentials.amount == null || amount == allAmount
-        ? allAmountFee
-        : calculateEstimatedFee(transactionCredentials.priority, amount);
+    var credentialsAmount = 0;
+    var amount = 0;
+    var fee = 0;
+
+    if (sendItemList.length > 1) {
+      final sendAllItems = sendItemList.where((item) => item.sendAll).toList();
+
+      if (sendAllItems?.isNotEmpty ?? false) {
+        throw BitcoinTransactionWrongBalanceException(currency);
+      }
+
+      final nullAmountItems = sendItemList.where((item) =>
+      stringDoubleToBitcoinAmount(item.cryptoAmount.replaceAll(',', '.')) <= 0)
+          .toList();
+
+      if (nullAmountItems?.isNotEmpty ?? false) {
+        throw BitcoinTransactionWrongBalanceException(currency);
+      }
+
+      credentialsAmount = sendItemList.fold(0, (previousValue, element) =>
+      previousValue + stringDoubleToBitcoinAmount(
+          element.cryptoAmount.replaceAll(',', '.')));
+
+      if (credentialsAmount > allAmount) {
+        throw BitcoinTransactionWrongBalanceException(currency);
+      }
+
+      amount = allAmount - credentialsAmount < minAmount
+          ? allAmount
+          : credentialsAmount;
+
+      fee = amount == allAmount
+          ? allAmountFee
+          : calculateEstimatedFee(transactionCredentials.priority, amount,
+          outputsCount: sendItemList.length + 1);
```

Comment:
for multi destination case we doesn't support to send all in that implementation, so we need to avoid "allAmount" or "sendAll" for multi destination implementation